src/xdsl/dialects/builtin.py

```diff
@@ -159,19 +195,21 @@ class VectorAttr(ParametrizedAttribute):
     data = ParameterDef(ArrayOfConstraint(IntegerAttr))
 
     @staticmethod
-    def get(data: List[int]) -> VectorAttr:
-        data_attr = [IntegerAttr.get(d) for d in data]
-        return VectorAttr([ArrayAttr.get(data_attr)])
+    def from_int_list(data: List[int]) -> VectorAttr:
+        data_attr = [IntegerAttr.from_int_and_width(d) for d in data]
+        return VectorAttr([ArrayAttr.from_list(data_attr)])
+
+    @staticmethod
+    @builder
+    def from_list(data: List[Union[int, IntegerAttr]]) -> VectorAttr:
+        data_attr = [IntegerAttr.build(d) for d in data]
+        return VectorAttr([ArrayAttr.from_list(data_attr)])
 
 
 @irdl_attr_definition
 class Float32Type(ParametrizedAttribute):
```

Comment:
Not related to this PR, but why is this a `ParametrizedAttribute`?

Reply:
The short answer is that using `ParametrizedAttribute` provides implementation for parsing and printing.
The longer answer is that `Data` attributes are supposed to only encapsulate raw data, and that if at some point we create new tools using attributes, we would only have to handle the `Data` attributes in a different way. So I try to minimize the number of `Data` we are using.